osquery/sql/sqlite_string.cpp

```diff
@@ -152,6 +152,41 @@ static void ip4StringToDecimalFunc(sqlite3_context* context,
   }
 }
 
+/**
+ * @brief Convert a hexadecimal string to integer
+ */
+static void hexStringToDecimalFunc(sqlite3_context* context,
+                                   int argc,
+                                   sqlite3_value** argv) {
+  assert(argc == 1);
+
+  if (SQLITE_NULL == sqlite3_value_type(argv[0])) {
+    sqlite3_result_null(context);
+    return;
+  }
+
+  long value = std::strtoll((char*)sqlite3_value_text(argv[0]), nullptr, 16);
+  sqlite3_result_int64(context, value);
+}
+
+/**
+ * @brief Convert an integer to hexadecimal string
+ */
+static void decimalToHexStringFunc(sqlite3_context* context,
+                                   int argc,
+                                   sqlite3_value** argv) {
+  assert(argc == 1);
+
+  if (SQLITE_NULL == sqlite3_value_type(argv[0])) {
+    sqlite3_result_null(context);
+    return;
+  }
+
+  char str[65];
+  std::snprintf(str, 19, "%#x", sqlite3_value_int(argv[0]));
+  sqlite3_result_text(context, str, strlen(str), SQLITE_TRANSIENT);
```

Comment:
Should this be strnlen?

Reply:
I can actually get the length from `snprintf` so I'll do that.